src/main/webapp/app/lecture/wizard-mode/lecture-wizard-units.component.ts

```diff
@@ -0,0 +1,315 @@
+        }
+
+        const { description, name, releaseDate, updateNotificationText } = attachmentUnitFormData.formProperties;
+        const { file, fileName } = attachmentUnitFormData.fileProperties;
+
+        this.currentlyProcessedAttachmentUnit = this.isEditingLectureUnit ? this.currentlyProcessedAttachmentUnit : new AttachmentUnit();
+        const attachmentToCreateOrEdit = this.isEditingLectureUnit ? this.currentlyProcessedAttachmentUnit.attachment! : new Attachment();
+
+        if (this.isEditingLectureUnit) {
+            // breaking the connection to prevent errors in deserialization. will be reconnected on the server side
+            this.currentlyProcessedAttachmentUnit.attachment = undefined;
+            attachmentToCreateOrEdit.attachmentUnit = undefined;
+        }
+
+        let notificationText: string | undefined;
+
+        if (updateNotificationText) {
+            notificationText = updateNotificationText;
+        }
+
+        if (name) {
+            attachmentToCreateOrEdit.name = name;
+        }
+        if (releaseDate) {
+            attachmentToCreateOrEdit.releaseDate = releaseDate;
+        }
+        attachmentToCreateOrEdit.attachmentType = AttachmentType.FILE;
+        attachmentToCreateOrEdit.version = 1;
+        attachmentToCreateOrEdit.uploadDate = dayjs();
+
+        if (description) {
+            this.currentlyProcessedAttachmentUnit.description = description;
+        }
+
+        const formData = new FormData();
+        formData.append('file', file, fileName);
+        formData.append('attachment', objectToJsonBlob(attachmentToCreateOrEdit));
+        formData.append('attachmentUnit', objectToJsonBlob(this.currentlyProcessedAttachmentUnit));
+
+        (this.isEditingLectureUnit
+            ? this.attachmentUnitService.update(this.lecture.id!, this.currentlyProcessedAttachmentUnit.id!, formData, notificationText)
+            : this.attachmentUnitService.create(formData, this.lecture.id!)
+        ).subscribe({
+            next: () => {
+                this.onCloseLectureUnitForms();
+                this.unitManagementComponent.loadData();
+            },
+            error: (res: HttpErrorResponse) => {
+                if (res.error?.params === 'file' && res?.error?.title) {
+                    this.alertService.error(res.error.title);
+                } else {
+                    onError(this.alertService, res);
+                }
+            },
+        });
+    }
+
+    /**
+     * Called when all selected exercises were linked from the component
+     */
+    onExerciseUnitCreated() {
+        this.onCloseLectureUnitForms();
+        this.unitManagementComponent.loadData();
+    }
+
+    startEditLectureUnit(lectureUnit: LectureUnit) {
+        this.isEditingLectureUnit = true;
+
+        lectureUnit.lecture = new Lecture();
+        lectureUnit.lecture.id = this.lecture.id;
+        lectureUnit.lecture.course = this.lecture.course;
+
+        switch (lectureUnit.type) {
```

Comment:
We need some form of inheritance / polymorphism instead of relying on switch cases

Reply:
I moved most of the logic out of the switch statement so it's cleaner but I couldn't get rid of it completely because of the different form data types